### PR TITLE
Drop support for Atom, invoke editors directly

### PR DESCRIPTION
**Pull request by niik**
## Description

Atom has been deprecated for two years now so we've given folks still using it plenty of time to migrate to something else so it's time we drop it. As part of this work I've removed the ability to launch an editor through a `.cmd` file (i.e. executing through a shell) on Windows. This required some changes to the way we discover applications installed via the JetBrains Toolbox to resolve their executable directly. VS Code and its relatives are easy enough to launch directly with a modification of the relative path.

cc @tsvetilian-ty Since you contributed toolbox support in https://github.com/desktop/desktop/pull/15780

## Release notes

Notes:

### CODE REVIEW

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -582,41 +541,31 @@ async function findApplication(editor: WindowsExternalEditor) {
...
-      )
-      const exists = await pathExists(editorPathInToolbox)
-      if (exists) {
-        return editorPathInToolbox
-      }
+const getJetBrainsToolboxEditors = memoizeOne(async () => {
+  const re = /^JetBrains Toolbox \((.*)\)/
+  const editors = new Array<WindowsExternalEditor>()
+
+  for (const key of enumerateKeys(HKEY.HKEY_CURRENT_USER, uninstallSubKey)) {
```
Comment (sergiou87) [inline app/src/lib/editors/win32.ts]: Should we also check `wow64UninstallSubKey` like in the previous implementation?

**Review by sergiou87 — COMMENTED** [review]
Looks good!! And works as expected, only one question…

File: app/src/lib/editors/win32.ts (hunk 1 above)
Comment (niik) [reply app/src/lib/editors/win32.ts]: I have no idea if these things exist in 32 bit versions but yeah, if the old logic looked there we should as well. I've addressed this in https://github.com/desktop/desktop/pull/19744/commits/86a419a7bf9777ae6ccf1f73045529f072d25d3a

**Review by tidy-dev — APPROVED** [review]
Looks good!

**Comment by Oldman124** [comment]
M-kopa

**Pull request merged**